dec-no-dos/xorDecryptor.cpp

```diff
@@ -0,0 +1,168 @@
+#include "xorDecryptor.h"
+
+namespace xorDecryptor
+{
+	unsigned int XorDecryptor::standardKeyHash(int start, int end){
+		if (start == 0) {
+			return hash[end];
+		}
+
+		return hash[end] - hash[start - 1] * prime[end - start + 1];
+	}
+
+	XorDecryptor::XorDecryptor(std::vector<char> encryptedText, Parameters parameters):
+		validator(parameters)
+	{
+		this->encryptedText = encryptedText;

```

Comment:
Use initializer list for that: https://en.cppreference.com/w/cpp/language/initializer_list